acs-integration-tests/src/test/java/com/ge/predix/integration/test/PolicyEvaluationCachingIT.java

```diff
@@ -0,0 +1,519 @@
+                this.policyHelper.deletePolicySet(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        testPolicyName, this.acsZone1Headers);
+                this.privilegeHelper.deleteSubject(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        subject.getSubjectIdentifier(), this.acsZone1Headers);
+            }
+        }
+    }
+
+    /**
+     * This test makes sure that cached policy evaluation results are properly invalidated when an affected subject is
+     * updated.
+     */
+    @Test
+    public void testPolicyEvalCacheWhenSubjectChanges() throws Exception {
+        BaseSubject subject = MARISSA_V1;
+        PolicyEvaluationRequestV1 policyEvaluationRequest = this.policyHelper
+                .createEvalRequest(MARISSA_V1.getSubjectIdentifier(), "sanramon");
+        String endpoint = this.acsUrl;
+
+        String testPolicyName = null;
+        try {
+            this.privilegeHelper.putSubject(this.acsAdminRestTemplate, subject, endpoint, this.acsZone1Headers,
+                    this.privilegeHelper.getDefaultAttribute());
+            String policyFile = "src/test/resources/single-site-based-policy-set.json";
+            testPolicyName = this.policyHelper.setTestPolicy(this.acsAdminRestTemplate, this.acsZone1Headers, endpoint,
+                    policyFile);
+
+            ResponseEntity<PolicyEvaluationResult> postForEntity = this.acsAdminRestTemplate.postForEntity(
+                    endpoint + PolicyHelper.ACS_POLICY_EVAL_API_PATH,
+                    new HttpEntity<>(policyEvaluationRequest, this.acsZone1Headers), PolicyEvaluationResult.class);
+
+            Assert.assertEquals(postForEntity.getStatusCode(), HttpStatus.OK);
+            PolicyEvaluationResult responseBody = postForEntity.getBody();
+            Assert.assertEquals(responseBody.getEffect(), Effect.PERMIT);
+
+            this.privilegeHelper.putSubject(this.acsAdminRestTemplate, subject, endpoint, this.acsZone1Headers,
+                    this.privilegeHelper.getAlternateAttribute());
+
+            postForEntity = this.acsAdminRestTemplate.postForEntity(endpoint + PolicyHelper.ACS_POLICY_EVAL_API_PATH,
+                    new HttpEntity<>(policyEvaluationRequest, this.acsZone1Headers), PolicyEvaluationResult.class);
+
+            Assert.assertEquals(postForEntity.getStatusCode(), HttpStatus.OK);
+            responseBody = postForEntity.getBody();
+            Assert.assertEquals(responseBody.getEffect(), Effect.NOT_APPLICABLE);
+        } finally {
+            if (testPolicyName != null) {
+                this.policyHelper.deletePolicySet(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        testPolicyName, this.acsZone1Headers);
+                this.privilegeHelper.deleteSubject(this.acsAdminRestTemplate, this.zoneHelper.getZone1Url(),
+                        subject.getSubjectIdentifier(), this.acsZone1Headers);
+            }
+        }
+    }
+
```

Comment:
Should we test for delete subject and update multiple subjects (postMultipleSubjects in PrivilegeHelper)?